src/stactools/cli/commands/validate.py

```diff
@@ -47,6 +66,28 @@ def validate_command(href, recurse, links, assets):
     return validate_command
 
 
+def root_validate(validator: Optional[STACValidator], object: STACObject,
+                  root: Optional[STACObject], recurse: bool, links: bool,
+                  assets: bool) -> List[str]:
+    # A None for validator uses the currently registered STACValidator.
+
+    # Swap to a new validator and safely restore it when finished
+    @contextmanager
+    def modded(validator: STACValidator):
+        old_validator = RegisteredValidator.get_validator()
+        RegisteredValidator.set_validator(validator)
+        try:
+            yield
+        finally:
+            RegisteredValidator.set_validator(old_validator)
+
+    if validator:
+        with modded(validator):
+            return validate(object, root, recurse, links, assets)
+    else:
+        return validate(object, root, recurse, links, assets)
```

Comment:
As Pete mentioned we might not need a contextmanager, but as an FYI `contextlib.nullcontext` can be useful for removing the need for the `if / else`  in cases where you maybe need a contextmanager: https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext.